### PR TITLE
Updates admin achievements

### PR DESCRIPTION
**Pull request by Goldsteel**
[qol][featurerequest][administration]

Dorms asked for this. You can give people a trophy, a medal or a dunce cap. I could add more.

image

image

:cl:
 * rscadd: Admins can now give out gold medals and dunce caps as achievements.
 * tweak: Changes some of the text for achievements to better reflect what's going on.

### CODE REVIEW

**Comment by SonixApache** [comment]
god i wish it was just a nanoui screen at this point

do the medal show up in the post-round achievements list?

**Comment by Goldsteel** [comment]
It's on the list at least. Somehow it made it this far without being nano.

That is also a good question and I do not know, let's find out.

**Comment by Goldsteel** [comment]
image

Yeah, but it doesn't keep the icon, it just stays as the cup currently.

**Comment by Goldsteel** [comment]
I ended up basically changing how the achievements proc works, will update soon

**Comment by Goldsteel** [comment]
This is finished too

**Comment by Blithering** [comment]
Spare would make a good icon for achievements

**Comment by Goldsteel** [comment]
I can only add more achievement items if I made it a list. Possible, but clunky.

**Comment by Blithering** [comment]
If it's a bother no worries, just a thought. Looks good as is.

File: code/game/gamemodes/gameticker.dm [hunk 1]
```diff
@@ -700,13 +700,12 @@ var/datum/controller/gameticker/ticker
 	return text
 
 /datum/controller/gameticker/proc/achievement_declare_completion()
+	if((achievements.len == 0) || (achievements.len == null))
```
Comment (Zth--) [inline code/game/gamemodes/gameticker.dm]:
```suggestion
	if(!achievements)
```

File: code/game/gamemodes/gameticker.dm [hunk 2]
```diff
@@ -700,13 +700,12 @@ var/datum/controller/gameticker/ticker
...
 	var/text = "<br><FONT size = 5><b>Additionally, the following players earned achievements:</b></FONT>"
-	var/icon/cup = icon('icons/obj/drinks.dmi', "golden_cup")
-	end_icons += cup
-	var/tempstate = end_icons.len
-	for(var/winner in achievements)
-		text += {"<br><img src="logo_[tempstate].png"> [winner]"}
-
+	//ITEM, CKEY, MOB NAME, AWARD NAME, AWARD DESC
+	for(var/i = 1, i < achievements.len, i+=5)
+		text += {"<br>[bicon(achievements[i])] <b>[achievements[i+1]]</b> as <b>[achievements[i+2]]</b> won <b>[achievements[i+3]]</b>, <b>[achievements[i+4]]!</b>"}
```
Comment (Zth--) [inline code/game/gamemodes/gameticker.dm]: whyyyy

File: code/modules/admin/admin_verbs.dm [hunk 3]
```diff
@@ -950,34 +950,35 @@ var/list/admin_verbs_mod = list(
...
+	if(achoice == "Gold cup")
+		award = new /obj/item/weapon/reagent_containers/food/drinks/golden_cup(get_turf(winner))
+	if(achoice == "Gold medal")
+		award = new /obj/item/clothing/accessory/medal/gold(get_turf(winner))
+	if(achoice == "Dunce Cap")
+		award = new /obj/item/clothing/head/dunce_cap(get_turf(winner))
+	award.name = name
+	award.desc = desc
+	if(iscarbon(winner) && (winner.stat == CONSCIOUS))
+		winner.put_in_hands(award)
```
Comment (Zth--) [inline code/modules/admin/admin_verbs.dm]: Should probably spawn the award in get_turf(winner) if he's not conscious

what happens if you call put_in_hands on an unconscious guy? If it already spawns it in the floor then simply remove that CONSCIOUS check

File: code/game/gamemodes/gameticker.dm (hunk 1 above)
Comment (Goldsteel) [reply code/game/gamemodes/gameticker.dm]: i keep forgetting i can do this, ill fix it when i get back

File: code/game/gamemodes/gameticker.dm (hunk 2 above)
Comment (Goldsteel) [reply code/game/gamemodes/gameticker.dm]: this takes the list of achievements in lots of 5 as the item, the ckey, the name, the award name and the description and just does it for each achievement to print them out at round end. im open to alternatives

File: __DEFINES/global.dm [hunk 4]
```diff
@@ -302,7 +302,7 @@ var/global/list/volunteer_gladiators = list()
 var/global/list/ready_gladiators = list()
 var/global/list/never_gladiators = list()
 
-var/global/list/achievements = list()
+//var/global/list/achievements = list()
```
Comment (Zth--) [inline __DEFINES/global.dm]: remove this comment and I merge this

Comment (Goldsteel) [reply __DEFINES/global.dm]: done

**Pull request merged**